src/robot/model/body.py

```diff
@@ -222,15 +230,15 @@ class Body(BaseBody):
     pass
 
 
-class Branches(BaseBody):
+class Branches(BaseBody, Generic[C]):
     """A list-like object representing IF and TRY branches."""
     __slots__ = ['branch_class']
 
-    def __init__(self, branch_class, parent=None, items=None):
+    def __init__(self, branch_class: Type[C], parent=None, items=None):
```

Comment:
`branch_class` cannot be anything (i.e. generic). It actually only be `IfBranch` or `TryBranch`, but just using `BodyItem` could be fine as well.